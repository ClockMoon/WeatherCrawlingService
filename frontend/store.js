import createSagaMiddleware from "redux-saga";
import rootSaga from "./sagas";
import reducer from "./reducers";
import { applyMiddleware, compose, createStore } from "redux";

export default (initialState, options) => {
  if (options) {
    const sagaMiddleware = createSagaMiddleware();
    const middlewares = [sagaMiddleware];
    const enhancer =
      process.env.NODE_ENV === "production"
        ? compose(applyMiddleware(...middlewares))
        : compose(
            applyMiddleware(...middlewares),
            !options.isServer &&
              typeof window.__REDUX_DEVTOOLS_EXTENSION__ !== "undefined"
              ? window.__REDUX_DEVTOOLS_EXTENSION__()
              : f => f
          );
    const store = createStore(reducer, initialState, enhancer);
    store.sagaTask = sagaMiddleware.run(rootSaga);
    return store;
  }
  const sagaMiddleware = createSagaMiddleware();
  const middlewares = [sagaMiddleware];
  const enhancer = compose(applyMiddleware(...middlewares));
  const store = createStore(reducer, initialState, enhancer);
  store.sagaTask = sagaMiddleware.run(rootSaga);
  return store;
};
