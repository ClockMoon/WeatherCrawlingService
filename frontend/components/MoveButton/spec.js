import { shallow, mount } from "enzyme";
import { PreviousMoveButton } from "./index";
import { Provider } from "react-redux";
import store from "../../store";
describe("PreviousMoveButton 컴포넌트 테스트.", () => {
  it("PreviousMoveButton 컴포넌트가 렌더링됩니다.", () => {
    const previousMoveButton = shallow(<PreviousMoveButton />);
    const previousContainer = previousMoveButton.find(".previousContainer");
    expect(previousContainer.debug()).toMatchSnapshot();
  });
});
