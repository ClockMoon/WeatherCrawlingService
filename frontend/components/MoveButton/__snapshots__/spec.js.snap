// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviousMoveButton 컴포넌트 테스트. PreviousMoveButton 컴포넌트가 렌더링됩니다. 1`] = `
"<div className=\\"previousContainer\\" onClick={[Function: movePreviousPage]}>
  <div className=\\"previous\\">
    이전
  </div>
</div>"
`;
